OpenTerm/View/TerminalView.swift

```diff
@@ -181,14 +191,105 @@ class TerminalView: UIView {
+                
+                let provider = NSItemProvider()
+                provider.registerFileRepresentation(forTypeIdentifier: uti, fileOptions: .openInPlace, visibility: .all,
+                                                    loadHandler: { (completion) in
+
+                    // read url from source provider
+                    let _ = item.itemProvider.loadObject(ofClass: URL.self,
+                                                         completionHandler: { (reader, error) in
+                        completion(reader, true, error)
+                    })
+
+                    return nil
+                })
+                provider.suggestedName = String(filename)
+                
+                let dragItem = UIDragItem(itemProvider: provider)
+                
+                // use label of dragged text as preview
+                dragItem.previewProvider = { return self.previewForDrag(dragRequest: dragRequest) }
+                items.append(dragItem)
+            }
+        }
+        return items
+    }
 
 }
 
 extension TerminalView: UITextDropDelegate {
+    func textDroppableView(_ textDroppableView: UIView & UITextDroppable, proposalForDrop drop: UITextDropRequest) -> UITextDropProposal {
+        let proposal = UITextDropProposal(operation: UIDropOperation.copy)
+        proposal.useFastSameViewOperations = false
+        proposal.dropAction = .replaceSelection
+        return proposal
+    }
+    
+    func textDroppableView(_ textDroppableView: UIView & UITextDroppable, willPerformDrop drop: UITextDropRequest) {
+        textView.pasteDelegate = self
+    }
+    
+    func textDroppableView(_ textDroppableView: UIView & UITextDroppable,
+                           dropSessionDidEnd session: UIDropSession) {
+        // move cursor to end of document when finished with drop
+        let end = textView.endOfDocument
+        textView.selectedTextRange = textView.textRange(from: end, to: end)
+        textView.becomeFirstResponder()
+        textView.pasteDelegate = nil
+    }
+}
 
+extension TerminalView : UITextPasteDelegate {
+    func textPasteConfigurationSupporting(_ textPasteConfigurationSupporting: UITextPasteConfigurationSupporting,
+                                          transform item: UITextPasteItem) {
+        
+        let uti = item.itemProvider.registeredTypeIdentifiers.first ?? (kUTTypeData as String)
+        item.itemProvider.loadInPlaceFileRepresentation(forTypeIdentifier: uti,
+                                                        completionHandler: { url, _, error in
+            // default behaviour on error or non-file URL
+            guard url?.isFileURL ?? false else {
+                item.setDefaultResult()
+                return
+            }
+                                                            
+            let result = relative(filename: url!.path, to: FileManager.default.currentDirectoryPath)
```

Comment:
As part of my tabs change (PR #70), using the FileManager's currentDirectoryPath is no longer correct, as each tab has its own currentDirectoryURL. Depending on when these get merged, this call will need to change